### DISCUSSION TITLE
Internal layout of the data structure returned by meshgrid

### DISCUSSION THREAD

**Discussion by mmoelle1** [alandefreitas/matplotplusplus#207] [Q&A]
Dear all,

From the source code I see the internal layout of the data structure that is returned by
```cpp
auto [X, Y] = meshgrid(linspace(-5, +5, 40), linspace(-5, +5, 40));
```
is an `std::pair` of `std::vector<std::vector<double>>`s (aka `vector_2d`). 

I am generating my data externally (with LibTorch) and would like to avoid copying it into the `vector_2d` structure. Is there a way to wrap external data into a structure that is accepted by functions like `surf`, etc.? I see that an overload of the `surf` function accepts `IterableIterables` but I cannot get it to work.

Thanks and best regards,
Matthias

**Comment by alandefreitas** [comment]
https://github.com/alandefreitas/matplotplusplus/blob/b45015e2be88e3340b400f82637b603d733d45ce/source/matplot/util/common.h#L334
